sycl/source/detail/graph_impl.cpp

```diff
@@ -507,27 +509,29 @@ modifiable_command_graph::finalize(const sycl::property_list &) const {
 
 bool modifiable_command_graph::begin_recording(queue &RecordingQueue) {
   auto QueueImpl = sycl::detail::getSyclObjImpl(RecordingQueue);
+  if (QueueImpl) {
```

Comment:
Why this and not an `assert`?

Reply:
I am not sure if we should expect non-null QueueImpl from getSyclObjImpl().
Coverity mentioned that it is possibly a nullptr.

Reply:
Well, if it can be a `nullptr`, then we need a test for that scenario as part of this PR.

Reply:
It is not ideal to create a test for the internal API getSyclObjImpl().
I can change this condition to assert(QueueImpl);
Would that satisfy your request?

Reply:
Yes, I'd rather have assert/SIGSEGV then pretend that the code is supported only to fail much later in an obscure place.

Reply:
Done.